src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -99,16 +99,26 @@ private function addRoute(RouteCollection $routeCollection, string $resourceClas
             throw new RuntimeException('Either a "route_name" or a "method" operation attribute must exist.');
         }
 
-        if (isset($operation['controller'])) {
-            $controller = $operation['controller'];
-        } else {
-            $actionName = sprintf('%s_%s', strtolower($operation['method']), $collection ? 'collection' : 'item');
+        $collectionString = $collection ? 'collection' : 'item';
+        $controller = $operation['controller'] ?? null;
+
+        if (null === $controller) {
+            $actionName = sprintf('%s_%s', strtolower($operation['method']), $collectionString);
             $controller = self::DEFAULT_ACTION_PATTERN.$actionName;
         }
 
-        $path = '/'.$normalizedShortName;
-        if (!$collection) {
-            $path .= '/{id}';
+        if ($operationName !== strtolower($operation['method'])) {
+            $actionName = sprintf('%s_%s', $operationName, $collectionString);
+        }
+
+        $path = $operation['path'] ?? null;
```

Comment:
to make it more readable L114-L122 could be extracted in a method `getPath(array $operation): string` or `determinePath`